### PR TITLE
viaf: fix update 

### PR DESCRIPTION
**Pull request by rerowep**
* Fixes problems during VIAF updates.   
     * Dependencies updates

### CODE REVIEW

File: rero_mef/agents/api.py [hunk 1]
```diff
@@ -50,56 +52,110 @@ def delete(self, force=False, dbcommit=False, delindex=False):
...
-        if self.deleted and not mef_data.get('deleted'):
-            mef_data['deleted'] = self.deleted
+        mef_records = []
+        mef_pids = set()
+        mef_actions = {}
+        viaf_records = [viaf_record] if viaf_record else []
+        viaf_pids = {viaf_record.pid} if viaf_record else set()
+        # get all VIAF records
+        for viaf in AgentViafRecord.get_viaf(self):
+            viaf_pid = viaf.pid
```
Comment (zannkukai) [inline rero_mef/agents/api.py:73]: could be removed and simply use `viaf.pid`

File: rero_mef/agents/api.py [hunk 2]
```diff
@@ -50,56 +52,110 @@ def delete(self, force=False, dbcommit=False, delindex=False):
...
+                viaf_records.append(viaf)
+        if len(viaf_records) > 1:
+            current_app.logger.error(
+                f'MULTIPLE VIAF FOUND FOR: {self.name} {self.pid} | '
+                f'viaf: {", ".join([viaf.pid for viaf in viaf_records])}'
+            )
+        # get all VIAF associated MEF records.
+        for viaf in viaf_records:
+            for mef in AgentMefRecord.get_mef(viaf.pid, viaf.name):
+                mef_pid = mef.pid
```
Comment (zannkukai) [inline rero_mef/agents/api.py:85]: same here

File: rero_mef/agents/viaf/api.py [hunk 3]
```diff
@@ -204,7 +204,7 @@ def update_online(agent_class, pid, online):
             :param online: Try to get following agent types online.
             :return: Agent record and performed action.
             """
-            action = 'NOTONLINE'
+            action = Action.NOTONLINE
```
Comment (zannkukai) [inline rero_mef/agents/viaf/api.py:207]: Cosmetic : `NOT_ONLINE` ?

File: rero_mef/api.py [hunk 4]
```diff
@@ -55,6 +55,8 @@ class Action(Enum):
     DELETEAGENT = 'delete agent'
     VALIDATIONERROR = 'validation error'
     ERROR = 'error'
+    NOTONLINE = 'not online'
+    NOTFOUND = 'not found'
```
Comment (zannkukai) [inline rero_mef/api.py:59]: cosmetic: `NOT_ONLINE` and `NOT_FOUND`

File: rero_mef/api.py [hunk 5]
```diff
@@ -55,6 +55,8 @@ class Action(Enum):
     DELETEAGENT = 'delete agent'
     VALIDATIONERROR = 'validation error'
```
Comment (zannkukai) [inline rero_mef/api.py:56]: cosmetic : `DELETE_AGENT` and `VALIDATION_ERROR`

**Review by zannkukai — APPROVED** [review]

**Review by Garfield-fr — APPROVED** [review]

**Pull request merged**